python/tests/test_fast.py

```diff
@@ -646,6 +646,34 @@ def test_custom_kernel_strides(self):
             )
             self.assertTrue(mx.allclose(mx.exp(a) * 32, outputs["out"]))
 
+    @unittest.skipIf(not mx.metal.is_available(), "Metal is not available")
+    def test_custom_kernel_helper(self):
+        mx.random.seed(7)
+        a = mx.random.normal(shape=(3, 6))
+        kernel = mx.fast.metal_kernel(
+            name="basic",
+            includes="""
+            template <typename T>
+            T do_exp(T x) {
+                return exp(x);
+            }
+            """,
+            source="""
+                uint elem = thread_position_in_grid.x;
+                out1[elem] = do_exp(a[elem]);
+            """,
+        )
+        out = kernel(
+            inputs={"a": a},
+            grid=(4, 1, 1),
```

Comment:
This launches 4 threads for 18 values and the check below `a[:2, :2]` should fail. It doesn't because there is no `self.assertTrue` on `mx.allclose`. Same goes for some tests further up in this file.

Reply:
Thanks for catching this!